src/adagrams.js

```diff
@@ -1,15 +1,206 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const oringinalPool = [
+    "A",
+    "A",
+    "A",
+    "A",
+    "A",
+    "A",
+    "A",
+    "A",
+    "A",
+    "B",
+    "B",
+    "C",
+    "C",
+    "D",
+    "D",
+    "D",
+    "D",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "F",
+    "F",
+    "G",
+    "G",
+    "G",
+    "H",
+    "H",
+    "I",
+    "I",
+    "I",
+    "I",
+    "I",
+    "I",
+    "I",
+    "I",
+    "I",
+    "J",
+    "K",
+    "L",
+    "L",
+    "L",
+    "L",
+    "M",
+    "M",
+    "N",
+    "N",
+    "N",
+    "N",
+    "N",
+    "N",
+    "O",
+    "O",
+    "O",
+    "O",
+    "O",
+    "O",
+    "O",
+    "O",
+    "P",
+    "P",
+    "Q",
+    "R",
+    "R",
+    "R",
+    "R",
+    "R",
+    "R",
+    "S",
+    "S",
+    "S",
+    "S",
+    "T",
+    "T",
+    "T",
+    "T",
+    "T",
+    "T",
+    "U",
+    "U",
+    "U",
+    "U",
+    "V",
+    "V",
+    "W",
+    "W",
+    "X",
+    "Y",
+    "Y",
+    "Z",
+  ];
+
+  let hand = [];
+  let gamePool = oringinalPool.slice();
+
+  while (hand.length < 10) {
+    let letter = gamePool[Math.floor(Math.random() * gamePool.length)];
+
+    hand.push(letter);
+
+    let index = gamePool.indexOf(letter);
+    if (index > -1) {
+      gamePool.splice(index, 1);
+    }
+  }
+  return hand;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  let word = input.toUpperCase();
+  let copy = lettersInHand.slice();
+  let result = true;
+
+  for (let letter of word) {
+    if (copy.includes(letter)) {
+      let index = copy.indexOf(letter);
+      if (index > -1) {
+        copy.splice(index, 1);
+      }
+    } else {
+      result = false;
+    }
+  }
+  return result;
 };
 
 export const scoreWord = (word) => {
-  // Implement this method for wave 3
+  const onePoint = ["A", "E", "I", "O", "U", "L", "N", "R", "S", "T"];
+  const twoPoint = ["D", "G"];
+  const threePoint = ["B", "C", "M", "P"];
+  const fourPoint = ["F", "H", "V", "W", "Y"];
+  const fivePoint = ["K"];
+  const eightPoint = ["J", "X"];
+  const tenPoint = ["Q", "Z"];
```

Comment:
It would be more efficient to have the letters and their scores stored in an object. Looking up the value of a key in an object is O(1) time complexity. With these arrays, you need to look through every single letter to find the matching one you want.